### ISSUE TITLE
Zodios request method is returning a union of all the possible API's endpoints

### ISSUE THREAD

**Issue by rikbrown** [ecyrbe/zodios#292]
I am seeing some weird behaviour with typings, but I might be doing something stupid.

In short, I'm using `makeEndpoint` and `makeApi` to split my definitions up. 

However, the return type of `client.request` seems to be either `never` (when using `makeApi`) or a union of all of the possible response types (when not using it). Have a look at the sandbox below and hover over the variables to see the inferred return types.

Sandbox: https://codesandbox.io/s/silent-lake-2yzxnl?file=/src/index.ts

The actual code works, it just seems to be typing.

Am I messing something up in my definitions or is something else awry?

**Comment by rikbrown** [comment]
The "union of all types" behaviour also seems to happen if I forgo `makeEndpoint` but use `satisfies` instead e.g.

```
export const getSubredditEndpoint = {
  method: "get",
  path: "r/:displayName/about",
  response,
} as const satisfies ZodiosEndpointDefinition<GetSubredditResponse
```

**Comment by rikbrown** [comment]
And when using `apiBuilder` and the spread operator like in this issue's example: https://github.com/ecyrbe/zodios/issues/149#issuecomment-1255469067

So I think I'm probably doing something wrong but would love pointers.

**Comment by ecyrbe** [comment]
So i'll rename this issue as a bug for `request` returning a union

**Comment by rikbrown** [comment]
Much cleaner with aliases, thank you. I saw them mentioned in your doc but it wasn't clear exactly how to use them (didn't realise they turned into methods like that).

Thanks for the fast help. I am really enjoying using Zodios!

**Comment by ecyrbe** [comment]
Thank you for sponsoring zodios.

i'll update the sponsors list. Very much appeciated. This helps a lot for keeping docs website running and renewed.

**Comment by ecyrbe** [comment]
@rikbrown union response bug is now fixed on `@zodios/core` `v10.7.3`
